packages/app-project/src/screens/ClassifyPage/components/WorkflowAssignmentModal/WorkflowAssignmentModalContainer.js

```diff
@@ -1,52 +1,122 @@
-}) {
-  const showPrompt = useMemo(() => promptAssignment(currentWorkflowID), [currentWorkflowID, promptAssignment])
-  const [ active, setActive ] = useState(showPrompt)
-  const [ dismissedForSession, setDismissed ] = useState(false)
-
-  // TODO: integrate session storage
-  function onDismiss (event) {
-    setDismissed(event.target.checked)
-  }
+import { MobXProviderContext, observer } from 'mobx-react'
+import { Button, Box, CheckBox, Text } from 'grommet'
+import { Modal, PrimaryButton, SpacedText } from '@zooniverse/react-components'
+import { useRouter } from 'next/router'
+import { useTranslation } from 'next-i18next'
+import styled from 'styled-components'
 
-  function closeFn () {
-    setActive(false)
+import NavLink from '@shared/components/NavLink'
+
+const StyledNavLink = styled(NavLink)`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`
+
+function useStore() {
+  const { store } = useContext(MobXProviderContext)
+
+  return {
+    /** assignedWorkflowID is fetched every 5 classifications per user session */
+    assignedWorkflowID: store.user.personalization.projectPreferences.settings?.workflow_id,
+    /** This function determines if the user has an assigned workflow and verifies that workflow is active in panoptes */
+    promptAssignment: store.user.personalization.projectPreferences.promptAssignment,
   }
+}
+
+function WorkflowAssignmentModalContainer({ currentWorkflowID = '' }) {
+  const { assignedWorkflowID, promptAssignment } = useStore()
+
+  const { t } = useTranslation('screens')
+  const router = useRouter()
+  const owner = router?.query?.owner
+  const project = router?.query?.project
+  const url = `/${owner}/${project}/classify/workflow/${assignedWorkflowID}`
 
-  useEffect(() => {
-    if (showPrompt && !dismissedForSession) {
-      setActive(true)
+  /** Active state of the Modal */
+  const [active, setActive] = useState(false)
+  /** State synced with sessionStorage */
+  const [dismissedForSession, setDismissedForSession] = useState(false)
+  /** Visual/aria checkbox state */
+  const [checkboxChecked, setCheckboxChecked] = useState(false)
+
+  useEffect(function checkForDismissal() {
+    if (window.sessionStorage.getItem('workflowAssignmentModalDismissed')) {
+      setDismissedForSession(true)
+    } else {
+      setDismissedForSession(false)
     }
+  }, [])
+
+  useEffect(
+    function modalVisibility() {
+      const showPrompt = promptAssignment(currentWorkflowID)
 
-    return () => setActive(false)
-  }, [assignedWorkflowID, dismissedForSession, showPrompt])
-
-  if (assignedWorkflowID) {
-    return (
-      <WorkflowAssignmentModal
-        active={active}
-        assignedWorkflowID={assignedWorkflowID}
-        closeFn={closeFn}
-        dismiss={onDismiss}
-        dismissedForSession={dismissedForSession}
-      />
-    )
+      if (showPrompt && !dismissedForSession) {
+        setActive(true)
+      } else {
+        setActive(false)
+      }
```

Comment:
Statements have side effects, so let's use an expression here. 

```suggestion
      const isActive = (showPrompt && !dismissedForSession)
      setActive(isActive)
```